### ISSUE TITLE
Can I add externalLabels?

### ISSUE THREAD

**Issue by SamDecrock** [utkuozdemir/nvidia_gpu_exporter#60]
To distinguish between machines, can I add an external label when deploying this exporter?

**Comment by utkuozdemir** [comment]
I think this answers you question as well: https://github.com/utkuozdemir/nvidia_gpu_exporter/issues/61#issuecomment-1009197891

**Comment by garykuok** [comment]
Hello, after searching for lots of examples online, I still couldn't change the uuid name scraping to the hostname. The metrics returned seems are not proper json.

Say, for example I have the extra info:
nvidia_smi_gpu_info{driver_model_current="[N/A]",driver_model_pending="[N/A]",driver_version="460.27.04",name="GeForce RTX 3090",uuid=<some_id>,vbios_version="94.02.42.00.9A"}

I want to changed the scraping name list from UUID to the name of the GPU, so I have the config below:

relabel_configs:
            - source_labels: [__uuid__]
              regex: '.*(name=\".*\").*'
              replacement: '$1'
              target_label: name
This apparently does not work. What should I modify to fulfill my wish? Actually, there's an "instance" key from Grafana's web UI when querying, I wonder if it can even better to use the "instance" in the dropdown list instead of "name" (since machines could have the exact same type of GPU)?

Many thanks in advance.

Chon Kit

**Comment by utkuozdemir** [comment]
I think you are on the right track with the `relabel_configs` and `source_labels`. See this post: https://blog.freshtracks.io/prometheus-relabel-rules-and-the-action-parameter-39c71959354a

If what you want is to do it on Grafana level, an example join looks like this:
```
max(nvidia_smi_temperature_gpu * on(uuid) group_left(name) nvidia_smi_gpu_info{}) by (name)
```

I adapted the join query from this post: https://ypereirareis.github.io/blog/2020/02/21/how-to-join-prometheus-metrics-by-label-with-promql/

I hope this helps.

**Comment by garykuok** [comment]
Hello, thank you for your reply. In the end, I didn't do the steps you mention.
I managed use the hostname in dropdown list menu by
1. change the variable `label_values(nvidia_smi_index, uuid)` to `label_values(nvidia_smi_index, instance)`
2. replace all `uuid` to `instance` in the JSON Model

I think the way it works is Prometheus uses "$gpu" (the variable) to fetch data, once we changed the it to something else, the selection query also needs to be changed to match the returned value.
